File: MgDev/UnitTest/WebTier/DotNet/TestCommon/CommonTests.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;

namespace OSGeo.MapGuide.Test.Common
{
    //Defines tests outside of the SQLite-based test suite

    public interface IExternalTest
    {
        void Execute(IPlatformFactory factory, ITestLogger logger);
    }

    public interface IPlatformFactory
    {
        MgService CreateService(int serviceType);
        MgMapBase CreateMap(MgResourceIdentifier mapDefinition);
        MgMapBase CreateMap(string coordSys, MgEnvelope env, string name);
        MgLayerBase CreateLayer(MgResourceIdentifier resId);
    }

    public class CommonTests
    {
        public static int Execute(IPlatformFactory factory, ITestLogger logger, ref int testsRun)
        {
            int failures = 0;
            var types = typeof(IPlatformFactory).Assembly.GetTypes();
            foreach (var type in types)
            {
                if (typeof(IExternalTest).IsAssignableFrom(type) && type.IsClass)
                {
                    var test = (IExternalTest)Activator.CreateInstance(type);
                    try
                    {
                        logger.WriteLine("****** Executing platform test: " + type.Name + " *********");
                        Console.WriteLine("Executing external platform test: " + type.Name);
                        test.Execute(factory, logger);
                    }
                    catch (AssertException ex)
                    {
                        logger.WriteLine("Assertion failure: " + ex.Message);
                        Console.WriteLine("Assertion failure: " + ex.Message);
                        failures++;
                    }
                    catch (Exception ex)
                    {
                        logger.WriteLine("General failure: " + ex.ToString());
                        Console.WriteLine("General failure: " + ex.ToString());
                        failures++;
                    }
                    finally
                    {
                        testsRun++;
                    }
                }
            }
            return failures;
        }
    }
}

File: MgDev/UnitTest/WebTier/Java/src/org/osgeo/mapguide/test/CommonTests.java
package org.osgeo.mapguide.test;

import java.util.*;

import org.osgeo.mapguide.test.common.*;
import org.osgeo.mapguide.test.platform.*;

public class CommonTests
{
    public static int Execute(IPlatformFactory factory, ITestLogger logger, MutableInteger testsRun) {
        int failures = 0;
        Vector<IExternalTest> tests = new Vector<IExternalTest>();

        //--------------- BEGIN Test Registration -----------------//
        tests.add(new ByteReaderTest());
        tests.add(new ByteReaderMemoryConstructorTest());
        tests.add(new ByteReaderStringConstructorTest());
        tests.add(new BatchPropertyCollectionTest());
        tests.add(new ClassDefinitionCollectionTest());
        tests.add(new CoordinateCollectionTest());
        tests.add(new CurvePolygonCollectionTest());
        tests.add(new CurveRingCollectionTest());
        tests.add(new CurveSegmentCollectionTest());
        tests.add(new CurveStringCollectionTest());
        tests.add(new FeatureCommandCollectionTest());
        tests.add(new FeatureSchemaCollectionTest());
        tests.add(new GeometryCollectionTest());
        tests.add(new IntCollectionTest());
        tests.add(new LayerCollectionTest());
        tests.add(new LayerGroupCollectionTest());
        tests.add(new LinearRingCollectionTest());
        tests.add(new LineStringCollectionTest());
        tests.add(new MapCollectionTest());
        tests.add(new PropertyCollectionTest());
        tests.add(new PropertyDefinitionCollectionTest());
        tests.add(new StringCollectionTest());
        tests.add(new StringPropertyCollectionTest());
        //--------------- END Test Registration -------------------//

        for (IExternalTest test : tests)
        {
            try
            {
                String clsName = CommonUtility.GetTypeName(test);
                logger.WriteLine("****** Executing platform test: " + clsName + " *********");
                Console.WriteLine("Executing external platform test: " + clsName);
                test.Execute(factory, logger);
            }
            catch (AssertException ex)
            {
                logger.WriteLine("Assertion failure: " + ex.getMessage());
                Console.WriteLine("Assertion failure: " + ex.getMessage());
                failures++;
            }
            catch (Exception ex)
            {
                logger.WriteLine("General failure: " + CommonUtility.ExceptionToString(ex));
                Console.WriteLine("General failure: " + CommonUtility.ExceptionToString(ex));
                failures++;
            }
            finally
            {
                testsRun.increment();
            }
        }
        return failures;
    }
}